### ISSUE TITLE
Euclidean Loss a factor of 3 greater than it should be

### ISSUE THREAD

**Issue by PeterQFR** [BVLC/caffe#4420]
Hi, 
I'm comparing the output of my net and a label arrays in the range of 0.0 to 1.0 using the Euclidean Loss layer, its not necessarily a binary classification. However when using the built in Euclidean Loss Layer my losses are around 10^1 - 10 ^2. When I calculate the loss directly it shows values that are similar numerically but around 10^-2 which I think is more correct. For example the solver outputs:
```
I0706 11:01:18.892063  6661 solver.cpp:229] Iteration 7, loss = 12.3456
I0706 11:01:18.892113  6661 solver.cpp:245]     Train net output #0: loss = 12.3456 (* 1 = 12.3456 loss)
I0706 11:01:18.892125  6661 sgd_solver.cpp:106] Iteration 7, lr = 1e-06
```
but I calcuate: `0.0123579179918`

My code for calculation is as follows:
```
 input = self.solver.net.blobs['labels'].data
 convout = self.solver.net.blobs['conv5_1'].data
 calcloss = np.sum(np.fabs((input - convout).flatten()).transpose()*np.fabs((input-convout).flatten()))/(2 * input.size)
```
Where input array is the label array, and convout is the output of the last layer of my network. 

Is this a bug or a design decision?

**Comment by seanbell** [comment]
Your code divides by the total number of entries (`input.size`), while the euclidean loss layer divides by the batch size (`input.shape[0]`).

Also note that this is a more efficient way of computing it:
`calcloss = np.sum((input - convout).square())/(2 * input.shape[0])`

**Comment by seanbell** [comment]
Closing since I don't think this is a bug.  Feel free to open a new issue discussing whether an option should be given to the EuclideanLossLayer with a different normalization open.
